### DISCUSSION TITLE
Datasets or benchmarks to add to Polaris

### DISCUSSION THREAD

**Discussion by cwognum** [polaris-hub/polaris#87] [Ideas]
Hi everyone, 

Polaris is built to host any relevant dataset or benchmark in ML-based drug discovery. The best way to help us at this stage of development is to identify and upload to the Hub such high-quality datasets or benchmarks. This will not only add value to the Hub by hosting more relevant content, but will also inform our feature roadmap.

This GH discussion is meant to collect, share and triage ideas for benchmarks or datasets to add next. I will make sure to regularly update the overview down below.

_If Polaris grows, it might make sense to move this to GH issues with a dedicated GH board._

----

### How to add a dataset or benchmark?
If you're interested in adding a dataset or benchmark to the Polaris Hub, here are some pointers to get started: 

1. Before you start, make sure that you have the rights to redistribute the dataset or benchmark through Polaris. Polaris supports a subset of CC licenses and the MIT license. 
2. If your dataset is small in size and has no complex dtypes such as 3D structures of proteins or conformers, you can save all of it efficiently in a Pandas DataFrame. In this case, these tutorials are a great way to get started: [Data Models](https://polaris-hub.github.io/polaris/main/tutorials/custom_dataset_benchmark.html) and [Data Curation](https://polaris-hub.github.io/auroris/stable/tutorials/getting_started.html)
3. If your data types do get more complex or if the datasets get large (>10GB), you will want to create a [Zarr](https://zarr.readthedocs.io/en/stable/) dataset. In this case, these tutorials are a great way to get started: [Zarr Datasets](https://polaris-hub.github.io/polaris/stable/tutorials/dataset_zarr.html) and [Dataset Factory](https://polaris-hub.github.io/polaris/stable/tutorials/dataset_factory.html).
4. Finally, since dataset curation can involve subjective decisions, it's important to be transparent about how a dataset or benchmark has been created. To this end, we've started the [Polaris Recipes](https://github.com/polaris-hub/polaris-recipes/) repository, and we invite you to create a PR there that can be run to reproduce the same dataset or benchmark. 

Let us know if you get stuck on any of the above! Our aim is to make this process as seamless as possible and your feedback is invaluable in improving our documentation.

----

### Datasets or benchmarks to add

_Last updated: July 17, 2024_

<details><summary>Completed</summary>
<p>

- [X] [Biogen ADMET dataset](https://doi.org/10.1021/acs.jcim.3c00160) - Thanks @zhu0619 
- [X] [PKIS-1 (Kinase Panel)](https://pubmed.ncbi.nlm.nih.gov/26501955) - Thanks @zhu0619 
- [X] [PKIS-2 (Kinase Panel)](https://doi.org/10.1371/journal.pone.0181585) - Thanks @zhu0619 
- [X] [Protein Gym](https://proteingym.org/) - Thanks to @danieldritter

</p>
</details> 

**In progress**
- [x] [ADME@NCATS](https://opendata.ncats.nih.gov/adme/) - In progress by @stwhitfield 
- [x] [D3R](https://drugdesigndata.org/about/datasets) - In progress by @stwhitfield 
- [x] [AstraZeneca ADME](https://www.ebi.ac.uk/chembl/document_report_card/CHEMBL3301361/) - In progress by @stwhitfield
- [ ] [MoleculeACE](https://pubs.acs.org/doi/10.1021/acs.jcim.2c01073)  - In progress by @derekvantilborg
- [ ] [CHORISO](https://arxiv.org/abs/2312.09004)  - In progress by @doncamilom
- [ ] [BEND](https://arxiv.org/abs/2311.12570) - In progress by @frederikkemarin
- [ ] [MARCEL](https://doi.org/10.48550/arXiv.2310.00115) - In progress by @zhu0619 
- [x] [Novartis Surrogate CYPP450 dataset](https://doi.org/10.1021/acs.chemrestox.3c00305) - In progress by @zhu0619  
- [ ] [PDE10A](https://pubmed.ncbi.nlm.nih.gov/36153472/) - In progress by @dessygil   

**Backlog**
- [ ] [ExCAPE-DB](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-017-0203-5)

**Comment by dessygil** [comment]
@timkucera I was wondering if I could take a swing at adding the datasets on ProteinShake?

**Comment by LFhase** [comment]
May I ask how to create and upload a customized dataset? It keeps responding " Client error '401 Unauthorized' for URL"

**Reply by cwognum** [reply]
Hey @LFhase, thanks for contributing to Polaris!

Have you logged in to Polaris yet? If so, could you try overwriting your login details using:

```
polaris login --overwrite
```

If this doesn't work, please file a bug report (there's an issue template for this in this repo) and I'll investigate!

**Reply by LFhase** [reply]
Hi @cwognum , thank you very much for your prompt reply! I have tried the login command and the issue remains with another error report. The details are given in the bug report https://github.com/polaris-hub/polaris/issues/145 . Thank you very much for helping look into it!

**Comment by Dinesh431786** [comment]
### Proposal: a geroscience benchmark — lifespan extension from structure

Polaris currently has no geroscience task. The datasets are potency, ADME and tox; none of them asks whether a molecule extends lifespan. I would like to propose one.

**Data.** 754 compounds curated from [DrugAge](https://genomics.senescence.info/drugs/) (Human Ageing Genomic Resources), CC BY 4.0, so redistributable with attribution. Target is the maximum reported percentage lifespan change, aggregated across *C. elegans*, *Drosophila*, mouse and yeast. It is a noisy label — that is a property of the field rather than the curation, and the dataset card says so plainly.

**What makes it worth adding** is not the data alone but the split design. Each compound carries three split assignments so the same model is scored three ways:

| column | split |
|---|---|
| `split_random` | seeded uniform hold-out |
| `split_scaffold` | Bemis-Murcko scaffold-disjoint |
| `split_frontier` | most-isolated Butina clusters assigned to test |

**The baseline result is negative, and that is the point.** A similarity-weighted 5-NN over Morgan(r=2, 2048) scores +0.063 / +0.153 / +0.098 Spearman across those three splits — all within roughly one standard error of each other and of zero (~150 test compounds gives SE ≈ 0.08).

We found that the hard way. An earlier curation pass produced a clean random > scaffold > frontier gradient that looked like a textbook generalization-collapse result. Three successive curation passes produced **three different orderings of the same three splits**. An effect whose sign flips with curation is noise, so the card instructs users not to report a point estimate without an error bar.

**Two curation decisions that change results,** both documented in the card:

- **Metal salts are not desalted.** A standard `FragmentParent` step merges lithium chloride (a GSK-3-mediated lifespan extender) with sodium chloride, methylmercury acetate with acetic acid, and vanadyl sulfate with sulfuric acid — for these compounds the metal is the pharmacology. Organic drug salts are desalted against an explicit counterion allow-list; metal-containing entries are left intact.
- **A `chem_class` column** separates `organic` (701) from `metal_complex`, `inorganic` and `small_organic`. Metal and elemental compounds are isolated in fingerprint space for reasons unrelated to medicinal chemistry, and including them changes which split appears hardest.

21 duplicate structures filed under different names were merged on InChIKey.

Curation script, curated data and a manifest of all counts and baseline scores: https://github.com/Dinesh431786/drugage-lifespan-benchmark — `python polaris_benchmark.py` regenerates the CSV and manifest from source.

Happy to contribute a recipe notebook to `polaris-recipes`. One practical question: since `upload_dataset` and `upload_benchmark` were removed from the client, what is the current route to get a dataset onto the Hub?
